aiida/storage/sqlite_temp/backend.py

```diff
@@ -195,11 +203,122 @@ def get_info(self, detailed: bool = False) -> dict:
         # results['repository'] = self.get_repository().get_info(detailed)
         return results
 
-    def bulk_insert(self, entity_type: EntityTypes, rows: list[dict], allow_defaults: bool = False) -> list[int]:
-        raise NotImplementedError
+    @staticmethod
+    @functools.lru_cache(maxsize=18)
+    def _get_mapper_from_entity(entity_type: EntityTypes, with_pk: bool):
```

Comment:
Where does this method, `bulk_insert` and `bulk_update` come from all of a sudden. Why do they need to be implemented here all of a sudden when just changing the repo backend? From a quick look I don't see any repo specific code in these implementations, right?

Reply:
these methods are currently only required by the `create_archive` and `import_archive` functions, providing a backend agnostic way to perform these operations (obviously more performant than the typical singular insert/update).
So a backend will run fine in "normal" operations without them, but will fail for archive creation/import.

Yes the code is specific to a backend using sqlalchemy to interface with a database. 
The code is basically copied from `PsqlDosBackend`, since obviously they are both using sqlalchemy (just sqlite instead of postgres)

Reply:
Haven't looked into the code in detail, but would it then maybe not almost make more sense to have `SqliteTempBackend` subclass `PsqlDosBackend` and then override certain methods? Probably would just have to change a few methods just to change the repository backend and some other minor things.